httpx/http_service.go

```diff
@@ -47,16 +69,12 @@ func NewServiceClient(serviceName string) *Client {
 		Transport: &RequestIDTransport{
 			Transport: &RetryTransport{
 				Retrier: retrier,
-				Transport: &Transport{Client: &http.Client{
-					Transport: &http.Transport{
-						Proxy: http.ProxyFromEnvironment,
-						Dial: (&net.Dialer{
-							Timeout:   15 * time.Second,
-							KeepAlive: 90 * time.Second,
-						}).Dial,
-						TLSHandshakeTimeout: 3 * time.Second,
-					},
-				}},
+				MethodsToRetry: map[string]bool {
+					"": true,
```

Comment:
I've never seen a caller do this. The documentation in request.go has a comment that says "For client requests an empty string means GET", so I treated an empty string the same way I treated GET.